### PR TITLE
Disallow saving branches of foreign scene child nodes

### LINKED ISSUE #56424: Saving editable child of a foreign scene should not be allowed and results in corruption
### Godot version

9f058674acebf9a7a0704cdbc26dc9f6d853b16e

### System information

Windows 10, Vulkan, Intel® Core™ i7-8700K, Nvidia GTX 1080 (471.11)

### Issue description

If you instanced or inherited a scene, you can select 'Save Branch as Scene' which will be allowed, but will result in corruption of the original scene since saving a branch automatically set the branch which was saved to an instance of the new scene, but since this is operating on foreign nodes, this should not be permitted.

### Steps to reproduce

Create a scene with a child node, save it, create another scene, instance the original scene as a child of the new scene, right click on it and select 'editable children', then right click on grey-labelled child node and select 'Save Branch as Scene'.

Alternatively, once you've created the first scene, right click on it in the file browser, select 'New Inherited Scene', and then attempt to right click on a child node and select 'Save Branch as Scene', the same issue will happen.

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by SaracenOne**
Pops up an error if attempting to save a branch as a new scene if it is the child node of either an instanced or inherited scene.
Closes #56424

### CODE REVIEW

**Comment by akien-mga** [comment]
Same comment as on 3.x PR.

File: editor/scene_tree_dock.cpp [hunk 1]
```diff
@@ -942,6 +942,18 @@ void SceneTreeDock::_tool_selected(int p_tool, bool p_confirm_override) {
 				break;
 			}
 
+			if (tocopy->get_owner() != scene) {
+				accept->set_text(TTR("Can't save a branch which is the child of an already instantiated scene. \nTo save this branch into its own scene, open the original scene, right click on this branch, and select 'Save Branch as Scene'."));
```
Comment (KoBeWi) [inline editor/scene_tree_dock.cpp:946]:
```suggestion
				accept->set_text(TTR("Can't save a branch which is a child of an already instantiated scene. \nTo save this branch into its own scene, open the original scene, right click on this branch, and select 'Save Branch as Scene'."));
```
Also probably
```suggestion
				accept->set_text(TTR("Can't save a branch which is the child of an already instantiated scene. \nTo save this branch into its own scene, open the original scene, right click on this branch, and select Save Branch as Scene."));
```
(seeing as above there are no quotes)

**Review by KoBeWi — APPROVED** [review]

File: editor/scene_tree_dock.cpp (hunk 1 above)
Comment (akien-mga) [reply editor/scene_tree_dock.cpp:946]: I'd keep the quotes, though maybe making them double (`\"Save Branch as Scene\"`).

**Comment by SaracenOne** [comment]
I've pushed an update with changes recommended by @koBeWi and @akien-mga. I'm not sure which is the best approach to go for the bit about 'Save Branch as Scene', but I went with Akien's suggestion for double-quotation for now, but if it's decided without quotation is more consistent with other messages, we can switch to that.

That being said, I did have a conversation with someone showing off some of the existing error messages, and it did make me think whether a proposal of more broadly overhauling Godot's style of message language more broadly could be useful for onboarding.

**Review by akien-mga — APPROVED** [review]

**Comment by SaracenOne** [comment]
Just pushed one more change quick change to both PRs which is to fix a minor typo I spotted

**Comment by akien-mga** [comment]
Will need a rebase to fix CI (temporary issue after renaming https://github.com/godotengine/regression-test-project).

**Comment by SaracenOne** [comment]
@akien-mga Rebased onto master now

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**